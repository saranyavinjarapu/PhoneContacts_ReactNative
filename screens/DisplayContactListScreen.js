import React from "react";
import { StyleSheet, View, FlatList } from "react-native";
import ContactItem from "../components/ContactItem";

const DisplayContactListScreen = (props) => {
  const selectContactHandler = (contactId) => {
    const selectedContact = props.contacts_data.filter(
      (data) => data.id == contactId
    );

    props.showSelectedContact(selectedContact);
  };
  return (
    <View style={styles.screen}>
      <FlatList
        keyExtractor={(item, index) => index.toString()}
        data={props.contacts_data}
        renderItem={(itemData) => (
          <ContactItem
            contactItem={itemData.item}
            onSelect={selectContactHandler}
          ></ContactItem>
        )}
      ></FlatList>
    </View>
  );
};

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    padding: 10,
  },
});

export default DisplayContactListScreen;
