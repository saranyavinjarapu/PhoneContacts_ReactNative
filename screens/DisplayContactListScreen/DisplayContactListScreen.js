import React from "react";
import { View, FlatList } from "react-native";
import styles from "./DisplayContactListScreenStyles";
import ContactItem from "../../components/ContactItem/ContactItem";

const DisplayContactListScreen = (props) => {
  const selectContactHandler = (contactId) => {
    const selectedContact = props.contacts_data.filter(
      (data) => data.id == contactId
    );
    props.showSelectedContact(selectedContact);
  };
  return (
    <View style={styles.screen}>
      <FlatList
        testID="flat_list"
        keyExtractor={(item, index) => index.toString()}
        data={props.contacts_data}
        renderItem={(itemData) => (
          <ContactItem
            contactItem={itemData.item}
            onSelect={selectContactHandler}
          ></ContactItem>
        )}
      ></FlatList>
    </View>
  );
};

export default DisplayContactListScreen;
