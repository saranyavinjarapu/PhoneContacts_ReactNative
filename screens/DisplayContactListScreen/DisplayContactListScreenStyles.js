import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    padding: 10,
  },
});

export default styles;
