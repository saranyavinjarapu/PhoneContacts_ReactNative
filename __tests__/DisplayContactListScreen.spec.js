import React from "react";
import { render, fireEvent } from "@testing-library/react-native";
import DisplayContactListScreen from "../screens/DisplayContactListScreen";

describe("Display Contacts List Screen Renders Correctly", () => {
  it("renders correctly", () => {
    const tree = render(<DisplayContactListScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should flatlist render data correctly", () => {
    const { queryByText, getByTestId } = render(<DisplayContactListScreen />);
    const flat_list_data = getByTestId("flat_list");
    expect(flat_list_data).not.toBeNull();
  });
});
