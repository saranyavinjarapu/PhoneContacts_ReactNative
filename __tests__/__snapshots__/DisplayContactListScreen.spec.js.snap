// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Contacts List Screen Renders Correctly renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "padding": 10,
    }
  }
>
  <RCTScrollView
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    testID="flat_list"
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
</View>
`;
